### DISCUSSION TITLE
AUR package for precompiled binary: rtx-bin

### DISCUSSION THREAD

**Discussion by tombh** [jdx/mise#440] [Ideas]
See: https://aur.archlinux.org/packages/rtx-bin

I'm a long time Arch user (BTW), but haven't made many AUR packages. So I'm tentatively putting this out there to see if it's useful.

I think ultimately its version and SHA could be automatically updated with this repo's on release process.

**Comment by jdx** [comment]
forgive me for not being very knowledgeable about arch, but what is the benefit of using this over the [existing non-pre-compiled version](https://github.com/jdxcode/rtx#aur)? Does the compilation have to happen on your laptop?

Somehow I think I thought these were precompiled _for you_ like with Homebrew but was that a completely false assumption?

We could probably work on the [infra](https://github.com/jdxcode/rtx/blob/main/scripts/release-aur.sh) for automatically updating the other aur repo. It's a _real_ pain to test this but shouldn't be complicated.

**Comment by tombh** [comment]
Exactly, Arch's AUR is the user-submitted side of the repos, it doesn't have any build infrastructure itself. So unless an AUR package contains the pre-compiled binary then, as you say, the compilation happens on your laptop.

It's worth noting that the AUR is also the most common way for packages to eventually reach Arch's official repos. The AUR keeps track of popularity, stability etc. So thanks to its presence in the AUR hopefully one day `rtx` will be an official Arch package anyway 😏

That infra script is great! It'll be easy to update this `rtx-bin` package if it proves to be in demand.

But first I'd be interested to see if I've done it right, and if anybody is even interested. I would have thought the pre-compiled version would be more popular, but that's just an assumption on my part.

**Comment by tombh** [comment]
Somebody noticed that I didn't update to the latest version and flagged the package. So at least one other person is using it.
